gator/checks/check_CountMarkdownTags.py

```diff
@@ -0,0 +1,103 @@
+"""Check that a Markdown file in a directory contains a specified number of Markdown tags."""
+
+import argparse
+
+from gator import checkers
+from gator import invoke
+
+
+def get_parser():
+    """Get a parser for the arguments provided on the command-line."""
+    # create the parser with the default help formatter
+    # use a new description since this is a stand-alone check
+    parser = argparse.ArgumentParser(
+        prog="CountMarkdownTags",
+        description="Check Provided by GatorGrader: CountMarkdownTags",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    # Required Named Checker Arguments {{{
+
+    required_group = parser.add_argument_group("required checker arguments")
+
+    # TAG: the markdown tag that must exist in the file
+    # REQUIRED? Yes
+    # https://github.com/readthedocs/commonmark.py
+    required_group.add_argument(
+        "--tag", type=str, help="markdown tag that exists in a file", required=True
+    )
+
+    # FILE: the file
+    # REQUIRED? Yes
+    required_group.add_argument(
+        "--file", type=str, help="file for checking", required=True
+    )
+
+    # DIRECTORY: the directory containing the file
+    # REQUIRED? Yes
+    required_group.add_argument(
+        "--directory",
+        type=str,
+        metavar="DIR",
+        help="directory with file for checking",
+        required=True,
+    )
+
+    # COUNT: the number of lines of output
+    # REQUIRED? Yes
+    required_group.add_argument(
+        "--count",
+        type=int,
+        metavar="COUNT",
+        help="how many tag instances should exist",
+        required=True,
+    )
+
+    # }}}
+
+    # Optional Named Checker Arguments {{{
+
+    optional_group = parser.add_argument_group("optional check arguments")
+
+    # EXACT: perform exact checking for commit counts (i.e,. "==" instead of ">=")
+    # REQUIRED? No
+    optional_group.add_argument(
+        "--exact",
+        help="equals instead of a minimum number",
+        default=False,
+        action="store_true",
+    )
+
+    # }}}
+
+    # add an epilog to list some of the available tags
+    # including those checks that are internal and user-provided
+    parser.epilog = "examples of available tags:\r\n  code, heading, image, link, list, paragraph\r\n  reference: https://spec.commonmark.org/0.29/"
```

Comment:
Should probably not use `\r\n` here and instead use `os.linesep` or something similar.

Reply:
Yes, I agree! I should also point out that your comment made me realize that, in fact, there was another defect in this checker: it was using the wrong help formatter and thus all of the newline formatting would be stripped out by `argparse` before it was displayed to the user.

I went ahead and fix this issue and the other related issues surfaced by your comment.